guest-agents/xen-guest-agent/pkg.yaml

```diff
@@ -0,0 +1,37 @@
+name: xen-guest-agent
+variant: alpine
+shell: /bin/sh
+install:
+  - cargo
+  - xen-dev
+  - clang-dev
+  - llvm-dev
+  - pkgconf
+steps:
+  - env:
+      PKG_CONFIG_PATH: /usr/lib/pkgconfig
+  - sources:
+      - url: https://gitlab.com/xen-project/xen-guest-agent/-/archive/{{ .XEN_GUEST_AGENT_VERSION }}/xen-guest-agent-{{ .XEN_GUEST_AGENT_VERSION }}.tar.gz
+        sha256: 492ba2022b4155b38443ddf34a5bdbb0d72966d0936aab8545b4a135fa69c990
+        sha512: 527611f1d0c4fec81ac44434304f426bb9ebe75ceeaac84a417397ec845da6a225514d6ff8b9dc03b5c27d4fa3b59a47fd1a64276fbf6b492664244796ef483c
+        destination: xe-guest-agent.tar.gz
+    prepare:
+      - |
+        sed -i 's#$VERSION#{{ .VERSION }}#' /pkg/manifest.yaml
+      - |
+        tar -xzvf xe-guest-agent.tar.gz --strip-components=1
+    build:
+      -  cargo build -F static --release
+    install:
+      - |
+        containerRoot=/rootfs/usr/local/lib/containers/xen-guest-agent
+        mkdir -p "$containerRoot"/usr/local/lib
+        mv /usr/lib/libgcc* "$containerRoot/usr/local/lib/"
```

Comment:
hmm rust doesn't produce fully static binaries?

Reply:
I think the cargo target needs to be `unknown-linux-gnu :thinking: for musl

Reply:
It should be something like `{ARCH}-unkown-linux-musl` right? If I target gnu, wouldn't it fail when running in Talos?

https://rust-lang.github.io/rfcs/1721-crt-static.html this might help. I'm checking it now

Reply:
ah yes, `unknown-linux-musl` i believe

Reply:
`+crt-static` worked. The binary is now standalone.